util/unifycr/src/unifycr.c

```diff
@@ -0,0 +1,269 @@
+"  -m, --mount=<path>        mount unifycr at <path>\n"
+"  -i, --transfer-in=<path>  stage in file(s) at <path>\n"
+"  -o, --transfer-out=<path> transfer file(s) to <path> on termination\n"
+"\n"
+"Available options for \"terminate\":\n"
+"  -c, --cleanup             clean up the unifycr storage on termination\n"
+"\n";
+
+static int debug;
+
+static void usage(int status)
+{
+    printf(usage_str, program);
+    exit(status);
+}
+
+static const char *basename(const char *path)
+{
+    const char *str = strrchr(path, '/');
+
+    if (str)
+        return &str[1];
+    else
+        return path;
+}
+
+static void parse_cmd_arguments(int argc, char **argv)
+{
+    int ch = 0;
+    int optidx = 2;
+    int cleanup = 0;
+    unifycr_cm_t consistency = UNIFYCR_CM_INVALID;
+    char *mountpoint = NULL;
+    char *transfer_in = NULL;
+    char *transfer_out = NULL;
+
+    while ((ch = getopt_long(argc, argv,
+                             short_opts, long_opts, &optidx)) >= 0) {
+        switch (ch) {
+        case 'c':
+            cleanup = 1;
+            break;
+
+        case 'C':
+            consistency = unifycr_read_consistency(optarg);
+            if (consistency < 0)
+                usage(1);
+            break;
+
+        case 'd':
+            debug = 1;
+            break;
+
+        case 'm':
+            mountpoint = strdup(optarg);
+            break;
+
+        case 'i':
+            transfer_in = strdup(optarg);
+            break;
+
+        case 'o':
+            transfer_out = strdup(optarg);
+            break;
+
+        case 'h':
+        default:
+            usage(0);
+            break;
+        }
+    }
+
+    opts_cmd.cleanup = cleanup;
+    opts_cmd.consistency = consistency;
+    opts_cmd.mountpoint = mountpoint;
+    opts_cmd.transfer_in = transfer_in;
+    opts_cmd.transfer_out = transfer_out;
+}
+
+int main(int argc, char **argv)
+{
+    int i = 0;
+    int ret = 0;
+    char *cmd = NULL;
+
+    program = basename(argv[0]);
+
+    if (argc < 2)
+        usage(1);
+
+    cmd = argv[1];
+
+    for (i = 0; i < sizeof(actions)/sizeof(char *); i++) {
+        if (strncmp(cmd, actions[i], strlen(actions[i])) == 0)
+            action = i;
+    }
+
+    if (action < 0)
+        usage(1);
+
+    parse_cmd_arguments(argc, argv);
+
+    if (debug) {
+        printf("\n## options from the command line ##\n");
+        printf("cleanup:\t%d\n", opts_cmd.cleanup);
+        printf("consistency:\t%s\n",
+               unifycr_write_consistency(opts_cmd.consistency));
+        printf("mountpoint:\t%s\n", opts_cmd.mountpoint);
+        printf("transfer_in:\t%s\n", opts_cmd.transfer_in);
+        printf("transfer_out:\t%s\n", opts_cmd.transfer_out);
+    }
+
+    ret = unifycr_read_resource(&resource);
+    if (ret) {
+        fprintf(stderr, "failed to detect a resource manager\n");
+        goto out;
+    }
+
+    if (debug) {
+        printf("\n## job allocation (%llu nodes) ##\n",
+               _llu(resource.n_nodes));
```

Comment:
For code readability I'd prefer an explicit typecast here, rather than a custom macro.

Reply:
I was lazy because it was a debug code :-)  I will fix.